kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver_with_constraints.h

```diff
@@ -0,0 +1,1180 @@
+        ProcessInfo &r_current_process_info = rModelPart.GetProcessInfo();
+        //contributions to the system
+        LocalSystemMatrixType relation_matrix = LocalSystemMatrixType(0, 0);
+        LocalSystemVectorType constant_vector = LocalSystemVectorType(0);
+        EquationIdVectorType slave_equation_ids = EquationIdVectorType(0);
+        EquationIdVectorType master_equation_ids = EquationIdVectorType(0);
+
+#pragma omp parallel for schedule(guided, 512) firstprivate(number_of_constraints, constraints_begin, relation_matrix, constant_vector, slave_equation_ids, master_equation_ids)
+        for (int i_constraints = 0; i_constraints < number_of_constraints; i_constraints++)
+        {
+            ModelPart::MasterSlaveConstraintContainerType::iterator it = constraints_begin + i_constraints;
+
+            //detect if the element is active or not. If the user did not make any choice the element
+            //is active by default
+            bool constraint_is_active = true;
+            if ((it)->IsDefined(ACTIVE))
+                constraint_is_active = (it)->Is(ACTIVE);
+
+            if (constraint_is_active)
+            {
+                //get the equation Ids of the constraint
+                it->EquationIdVector(slave_equation_ids, master_equation_ids, r_current_process_info);
+                //calculate constraint's T and b matrices
+                it->CalculateLocalSystem(relation_matrix, constant_vector, r_current_process_info);
+
+                //assemble the Constraint contribution
+                AssembleSlaves(slave_equation_ids, master_equation_ids, relation_matrix);
+            }
+        }
+        const double stop_formulate = OpenMPUtils::GetCurrentTime();
+        KRATOS_INFO_IF("ResidualBasedBlockBuilderAndSolverWithConstraints", (this->GetEchoLevel() >= 1 && rModelPart.GetCommunicator().MyPID() == 0)) << "Formulate global constraints time: " << stop_formulate - start_formulate << std::endl;
+
+        KRATOS_CATCH("ResidualBasedBlockBuilderAndSolverWithConstraints::FormulateGlobalMasterSlaveRelations failed ..");
+    }
+
+
+    /**
+     * @brief   this method assembles the given list of slave equation ids and corresponding master equatio ids
+     * @param   rSlaveEquationIdVector list of slave equation ids to be assembled.
+     * @param   rMasterEquationIdVector list of corresponding master equation ids.
+     * @param   rRelationMatrix the matrix relating the rSlaveEquationIdVector with rMasterEquationIdVector
+     */
+    void AssembleSlaves(EquationIdVectorType& rSlaveEquationIdVector, EquationIdVectorType& rMasterEquationIdVector, LocalSystemMatrixType& rRelationMatrix)
+    {
+        KRATOS_TRY
+        int slave_count = 0;
+        for (auto &slave_equation_id : rSlaveEquationIdVector)
+        {
+            /*int master_count = 0;
+            auto global_constraint = mGlobalMasterSlaveRelations.find(slave_equation_id);
```

Comment:
This particular comment : I left it there because I want to use PointerVectoreSet   once the issue with unique pointers and the PointerVectorSet is solved.

Reply:
ok, but try to avoid such things as much as possible.